packages/core-sdk/src/utils/registerHelper.ts

```diff
@@ -0,0 +1,1319 @@
+    });
+  }
+  if ("derivData" in request) {
+    const derivData = await validateDerivativeData({
+      derivativeDataInput: request.derivData,
+      rpcClient,
+      wallet,
+      chainId,
+    });
+    const totalDerivativeMintingFee = await calculateDerivativeMintingFee({
+      derivData,
+      rpcClient,
+      chainId,
+      wallet,
+    });
+    const requestWithDeriv = { ...baseRequest, derivData };
+
+    if ("royaltyShares" in request) {
+      return transferMintAndRegisterIpAndMakeDerivativeAndDistributeRoyaltyTokensRequest({
+        request: {
+          ...requestWithDeriv,
+          royaltyShares: request.royaltyShares,
+        },
+        nftMintFee,
+        isPublicMinting,
+        totalDerivativeMintingFee,
+        royaltyTokenDistributionWorkflowsClient,
+      });
+    }
+    return transferMintAndRegisterIpAndMakeDerivativeRequest({
+      request: {
+        ...requestWithDeriv,
+        derivData,
+      },
+      derivativeWorkflowsClient,
+      nftMintFee,
+      isPublicMinting,
+      totalDerivativeMintingFee,
+    });
+  }
+  throw new Error("Invalid mint and register request type");
+};
+export const validateLicenseTermsData = async (
+  licenseTermsData: LicenseTermsDataInput[],
+  rpcClient: PublicClient,
+): Promise<{
+  licenseTerms: LicenseTerms[];
+  licenseTermsData: LicenseTermsData[];
+}> => {
+  const licenseTerms: LicenseTerms[] = [];
+  const processedLicenseTermsData: LicenseTermsData[] = [];
+  for (let i = 0; i < licenseTermsData.length; i++) {
+    const licenseTerm = await validateLicenseTerms(licenseTermsData[i].terms, rpcClient);
+    const licensingConfig = validateLicenseConfig(licenseTermsData[i].licensingConfig);
+    if (licensingConfig.mintingFee > 0 && licenseTerm.royaltyPolicy === zeroAddress) {
+      throw new Error("A royalty policy must be provided when the minting fee is greater than 0.");
+    }
+    licenseTerms.push(licenseTerm);
+    processedLicenseTermsData.push({
+      terms: licenseTerm,
+      licensingConfig: licensingConfig,
+    });
+  }
+  return { licenseTerms, licenseTermsData: processedLicenseTermsData };
+};
+
+export const getRoyaltyShares = (royaltyShares: RoyaltyShare[]) => {
+  let actualTotal = 0;
+  let sum = 0;
+  const shares = royaltyShares.map((share) => {
+    if (share.percentage <= 0) {
+      throw new Error("The percentage of the royalty shares must be greater than 0.");
+    }
+    if (share.percentage > 100) {
+      throw new Error("The percentage of the royalty shares must be less than or equal to 100.");
+    }
+    sum += share.percentage;
+    if (sum > 100) {
+      throw new Error("The sum of the royalty shares cannot exceeds 100.");
+    }
+    const value = (share.percentage / 100) * royaltySharesTotalSupply;
+    actualTotal += value;
+    return { ...share, percentage: value };
+  });
+  return { royaltyShares: shares, totalAmount: actualTotal } as const;
+};
+
+export const validateDerivativeData = async ({
```

Comment:
this file is getting pretty big (+1k lines), maybe we move the validation helpers into a separate file? I think there are also other related utils in this file that we can group together and make separate files for.